src/main/java/appeng/util/item/FuzzyItemVariantList.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * This file is part of Applied Energistics 2.
+ * Copyright (c) 2013 - 2020, AlgorithmX2, All rights reserved.
+ *
+ * Applied Energistics 2 is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Applied Energistics 2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Applied Energistics 2.  If not, see <http://www.gnu.org/licenses/lgpl>.
+ */
+
+package appeng.util.item;
+
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.Map;
+
+import com.google.common.base.Preconditions;
+
+import net.minecraft.item.ItemStack;
+
+import it.unimi.dsi.fastutil.objects.Object2ObjectAVLTreeMap;
+import it.unimi.dsi.fastutil.objects.Object2ObjectSortedMap;
+
+import appeng.api.config.FuzzyMode;
+import appeng.api.storage.data.IAEItemStack;
+
+/**
+ * This variant list is optimized for damageable items, and supports selecting durability ranges with
+ * {@link #findFuzzy(IAEItemStack, FuzzyMode)}.
+ */
+class FuzzyItemVariantList extends ItemVariantList {
+
+    static final SharedStackComparator COMPARATOR = new SharedStackComparator();
+
+    // NOTE: We only use Object as they key here so we can pass our special DamageBounds to the subMap method.
+    // We NEVER put any keys in this map that are not AESharedItemStacks.
+    private final Object2ObjectSortedMap<Object, IAEItemStack> records = new Object2ObjectAVLTreeMap<>(COMPARATOR);
```

Comment:
We might also look into using `Object2ObjectRBTreeMap` as replacement here. AVL should be a bit faster for retrieval, but a bit more costly on insert. Hard to guess in advance really and also hard to reliably benchmark under real conditions.

The other hash based ones are not alternative as none implements `subMap()`

Reply:
Yeah, I'd leave it as-is for now and just observe.